oai_agents/agents/rl.py

```diff
@@ -302,33 +312,43 @@ def train_agents(self, total_train_timesteps, tag_for_returning_agent, exp_name=
 
         if self.checkpoint_rate is not None:
             if self.args.resume:
-                path = self.args.base_dir / 'agent_models' / experiment_name
-
-                ckpts = [name for name in os.listdir(path) if name.startswith("ck")]
-                ckpts_nums = [int(c.split('_')[1]) for c in ckpts]
-                sorted_idxs = np.argsort(ckpts_nums)
-                ckpts = [ckpts[i] for i in sorted_idxs]
-                self.ck_list = [(c[0], path, c[2]) for c in resume_ck_list] if resume_ck_list else [({k: 0 for k in self.args.layout_names}, path, ck) for ck in ckpts]
+                path = RLAgentTrainer.get_model_path(
+                    base_dir=self.args.base_dir,
+                    exp_folder=self.args.exp_dir,
+                    model_name=self.name
+                )
+                if not path.exists():
+                    print(f"Warning: The directory {path} does not exist.")
+                    self.save_init_model_and_cklist()
+                else:
+                    ckpts = [name for name in os.listdir(path) if name.startswith(KeyCheckpoints.CHECKED_MODEL_PREFIX)]
+                    if not ckpts:
+                        print(f"Warning: No checkpoints found in {path} with prefix '{KeyCheckpoints.CHECKED_MODEL_PREFIX}'.")
+                        self.save_init_model_and_cklist()
+                    else:
+                        ckpts_nums = [int(c.split('_')[1]) for c in ckpts]
+                        sorted_idxs = np.argsort(ckpts_nums)
+                        ckpts = [ckpts[i] for i in sorted_idxs]
+                        self.ck_list = [(c[0], path, c[2]) for c in resume_ck_list] if resume_ck_list else [
+                                ({k: 0 for k in self.args.layout_names}, path, ck) for ck in ckpts]
             else:
-                self.ck_list = []
-                path, tag = self.save_agents(tag=f'ck_{len(self.ck_list)}')
-                self.ck_list.append(({k: 0 for k in self.args.layout_names}, path, tag))
+                self.save_init_model_and_cklist()
+
 
         best_path, best_tag = None, None
 
         self.steps = self.start_step
-        curr_timesteps = self.start_timestep
-        prev_timesteps = self.learning_agent.num_timesteps
+        self.learning_agent.num_timesteps = self.n_envs*self.start_timestep
```

Comment:
Why do we need to change this?

Reply:
If n_env is 10, and the start_timestep is 100, then the timestep we count in RLagentlearner will be 1000. Therefore, to measure the num_timesteps for learning_agent, this line of code is necessary.

I assume most of us will be still confused here. I can clarify it in our meeting or we can set up a zoom meeting to specifically discuss it.